### ISSUE TITLE
Clarify picking the shop location process

### ISSUE THREAD

**Issue by serpico** [openfoodfacts/open-prices-frontend#568]
### Problem
Currently picking the shop location can be tricky.

For the sake of example look up **Leclerc Canteleu** in the search bar, out of the 7 results I can navigate on the map and isolate 3, all located Rue du canal but I can't match them individually with the choices offered on the left.

### Proposed solution
I thought maybe clicking on one blue tack would highlight the corresponding option/choice on the left but it didn't work that way.

Displaying, on the left ( where house is currently displayed ), amenity= or shop= value as stored in OSM could help in that case, being an OSM contributor/user I even thought displaying the OSM ID/OSM way ( 65403813 in that case ) would sort me out but that's not for everyone.

**Comment by serpico** [comment]
another test case

Input :  **Tang Frères, 48 avenue d'Ivry** 

1st pick : [2621636797](https://www.openstreetmap.org/node/2621636797)

2nd pick, after figuring the first was not the appropriate one, zooming on the map and using the blue tack to choose : [2573626630](https://www.openstreetmap.org/node/2573626630)

Actual location/Expected : [252814154](https://www.openstreetmap.org/node/252814154)

**Comment by raphodn** [comment]
Thanks for the feedback ! I dug into the code, there'is indeed some small issues after switching Nominatim to Photon.

Current search results for Leclerc Canteleu
![image](https://github.com/openfoodfacts/open-prices-frontend/assets/7147385/e3da5662-a3a3-4713-bf3e-2003c328ffd3)

After some fixes + improvements
![image](https://github.com/openfoodfacts/open-prices-frontend/assets/7147385/7bc32fcc-3a57-40db-a948-2f10c735f3a8)

Will open a PR :+1:

**Comment by serpico** [comment]
Thanks! displaying `osm_tag` really helps to clear the confusion.

**Comment by raphodn** [comment]
Yeah it was actually a bug, I was showing the wrong key from the Photon data..

**Comment by raphodn** [comment]
And I noted your other ideas in #37 ! Showing the OSM ID could be easy to do (as a settings options)

**Comment by raphodn** [comment]
fyi @serpico i did the osm id display option in #600 (in prod)

**Comment by serpico** [comment]
> fyi @serpico i did the osm id display option in #600 (in prod)

@raphodn Yeah I saw it during the WE I think thanks again!!

I thought about something else but I'm not sure it's useful or really improve the user experience that's why I'm reluctant to open a specific issue for it, so this is just in case you see it.

In the location search page when the user change the zoom level or the bounding box of the map, trim the results on the left of the page accordingly to display only the markers visible on the map, could make it it easier to pick the relevant location.

Example, in that case on the left side, the 5 initials results would be trimmed to the 3 corresponding to the markers on the map would be displayed on the left.
![trim results](https://github.com/openfoodfacts/open-prices-frontend/assets/1227077/21b7904d-823b-4c4e-b01a-cca7c4d0b45c)
